### PR TITLE
[W5.7][F11-4]Tan Zhi Wei 

### PR DESCRIPTION
**Pull request by ZhiWei94**
Add Tag Search Function

### CODE REVIEW

File: test/java/seedu/addressbook/commands/TagCommandTest.java [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+        assertTagCommandBehavior(new String[]{"test", "test"}, Arrays.asList(td.dan));
+
+        //Keyword matching a word in address: not matched
+        assertTagCommandBehavior(new String[]{"Clementi"}, Collections.emptyList());
+
+        assertTagCommandBehavior(new String[]{"classmates"}, Arrays.asList(td.elle, td.felicia));
+    }
+
+    /**
+     * Executes the find command for the given keywords and verifies
```
Comment (souless94) [inline test/java/seedu/addressbook/commands/TagCommandTest.java:46]: Should it be" Executes the tag command" instead of "Executes the find command"?

**Review by souless94 — COMMENTED** [review]
clarifying on Some words for the comments in TagCommandTest

File: src/seedu/addressbook/commands/TagCommand.java [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+    /**
+     * Retrieves all persons in the address book whose names contain some of the specified keywords.
+     *
+     * @param keywords for searching
+     * @return list of persons found
+     */
+    private List<ReadOnlyPerson> getPersonsWithNameContainingAnyTag(Set<String> keywords) {
+        final List<ReadOnlyPerson> matchedPersons = new ArrayList<>(); //create object matchedPersons
+
+        // For All persons in addressBook, starting from Index 1
```
Comment (jelneo) [inline src/seedu/addressbook/commands/TagCommand.java:49]: unnecessary comment

File: test/java/seedu/addressbook/commands/TagCommandTest.java [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+import seedu.addressbook.data.person.ReadOnlyPerson;
+import seedu.addressbook.util.TypicalPersons;
+
+public class TagCommandTest {
+
+    private final AddressBook addressBook = new TypicalPersons().getTypicalAddressBook();
+    private final TypicalPersons td = new TypicalPersons();
+
+    @Test
+    public void execute() throws IllegalValueException {
```
Comment (jelneo) [inline test/java/seedu/addressbook/commands/TagCommandTest.java:24]: Perhaps you can give  a more descriptive test method name?

**Review by jelneo — APPROVED** [review]
Good effort. Please close this PR after reading my comments.

**Pull request closed without merging**